mutation-testing/check-packages-and-shards/action.yml

```diff
@@ -0,0 +1,123 @@
+name: Check Packages and Shards
+
+outputs:
+  run_big_packages:
+    value: ${{ steps.check_packages_and_shards.outputs.run_big_packages }}
+  big_packages_with_shards:
+    value: ${{ steps.check_packages_and_shards.outputs.big_packages_with_shards }}
+  run_small_packages:
+    value: ${{ steps.check_packages_and_shards.outputs.run_small_packages }}
+  small_packages_with_shards:
+    value: ${{ steps.check_packages_and_shards.outputs.small_packages_with_shards }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout stacks-core repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install cargo-mutants
+      shell: bash
+      run: cargo install --version 23.12.2 cargo-mutants
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    - name: Update git diff
+      shell: bash
+      run: |
+        input_file="git.diff"
+        temp_file="temp_diff_file.diff"
+
+        # Check if the commands exist on the host
+        for cmd in tac awk sed; do
+          command -v "${cmd}" > /dev/null 2>&1 || echo "Missing command: ${cmd}"
+        done
+
+        # Reverse the file, remove 4 lines after '+++ /dev/null', then reverse it back (editors can't go backwards - to remove lines above)
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
```

Comment:
before these commands run, add a check that the expected files exist. if not, exit non-zero.
ex:
```
if [ ! -f "$input_file" ]; then
   echo "file missing"
   exit 1
fi
```